janusgraph-core/src/main/java/org/janusgraph/graphdb/relations/CacheVertexProperty.java

```diff
@@ -133,7 +133,7 @@ public byte getLifeCycle() {
 
     @Override
     public void remove() {
-        if (!tx().isRemovedRelation(super.longId())) {
+        if (!tx().isRemovedRelation(super.longId()) && !tx().getConfiguration().isReadOnly()) {
```

Comment:
Should trying to delete a relation not throw a PermanentStorageException on a read-only graph?